doc: Clarify the Repository lifecycle

FeynmanZhou opened the pull request:
fix #36

TheFoxAtWork reviewed (commented):
Looks good, please make sure your governance outlines a supermajority and is linked here.

SteveLasker commented:
Having the process outlined is goodness. Archiving a repo that's been dormant may not mean the project isn't useful for others to consume as reference. It may just be stable and doesn't need additional changes. In this case, archiving can indicate it's stable, and not taking additional changes, because there aren't any maintainers, or the project is simply stable. 

I'd also suggest adding text that a repo must have at least 30 days notice to its maintainers before archival is completed. 

Deleting a project is far more destructive and removes content that others may need. I'd suggest we clarify under what circumstances a project would be deleted, vs. archived.

yizha1 commented on REPOSITORIES.md:
```diff
@@ -0,0 +1,35 @@
+
+Archived repositories can be proposed for unarchiviation by opening a GitHub issue in the [notaryproject/.github](https://github.com/notaryproject/.github) repository. Once the proposal issue has a supermajority of approval from Notary Project governance maintainers, the archived repository can be changed to active. In general, the same rules as for new repositories apply. 
+
+### Removal
```
I suggest removing this process for now. IMO, it is very rare to request removing a repo rather than archiving. We can add this process later if there is really a case.

FeynmanZhou replied:
Removed this section

yizha1 reviewed (commented):
Overall looks good. Just one comment.

priteshbandi commented on REPOSITORIES.md:
```diff
@@ -0,0 +1,35 @@
+
+### Lifecycle change 
+
+A repository can be archived if the sub-project maintainers no longer maintain it or no activity for a long time. The sub-project maintainers of a given repository can propose changing its status by opening a GitHub issue in the [notaryproject/.github](https://github.com/notaryproject/.github) repository. The [Notary Project governance maintainers](MAINTAINERS) will take into account the communit ...
```
* Please elaborate on what does ` sub-project maintainers no longer maintain it` means?
* ` or no activity for a long time.` Can we quantify how long?

priteshbandi commented on REPOSITORIES.md:
```diff
@@ -0,0 +1,35 @@
+
+#### Archiviation
+
+Repositories showing little to no activity during the time span of a year can be proposed for Archiviation by opening a GitHub [issue](https://github.com/notaryproject/.github/issues). Once the proposal issue has a supermajority of approval from Notary Project governance maintainers, they can be archived. [Archived repositories](https://docs.github.com/en/repositories/archiving-a-github-repositor ...
```
Should also update readme file to reflect that project is in archived and no longer maintained?

priteshbandi replied:
Should we also update readme file to reflect that project is in archived and no longer maintained?

FeynmanZhou commented:
> Having the process outlined is goodness. Archiving a repo that's been dormant may not mean the project isn't useful for others to consume as reference. It may just be stable and doesn't need additional changes. In this case, archiving can indicate it's stable, and not taking additional changes, because there aren't any maintainers, or the project is simply stable.
> 
> I'd also suggest adding text that a repo must have at least 30 days notice to its maintainers before archival is completed.
> 
> Deleting a project is far more destructive and removes content that others may need. I'd suggest we clarify under what circumstances a project would be deleted, vs. archived.

Thanks @SteveLasker for the suggestions. I added a paragraph about the archival notice and removed the "Removal" section according to your inputs above.

priteshbandi reviewed (approved):
LGTM

yizha1 commented on REPOSITORIES.md:
```diff
@@ -0,0 +1,31 @@
+A new sub-project can be created as a repository and contributed to the Notary Project organization by opening a GitHub issue to request [Notary Project governance maintainers](MAINTAINERS) to vote in the
+[notaryproject/.github](https://github.com/notaryproject/.github) repository. A new sub-project can be created after having a supermajority of approval from Notary Project governance maintainers.
+
+If the decision is to add the proposed project, then one of the Notary Project GitHub organization admins will assist the issue opener in creating a repository or transferring an existing repository to the Notary Project organization. Upon creation, the repository must be reviewed to make sure it follows [Notary Governance](GOVERNANCE.md). 
```
```suggestion
If the decision is to add the proposed project, then one of the Notary Project GitHub organization admins will assist the issue opener in creating a repository or transferring an existing repository to the Notary Project organization. Upon creation, the repository must be reviewed to make sure it follows the [Notary Project Governance](GOVERNANCE.md). 
```

yizha1 commented on REPOSITORIES.md:
```diff
@@ -0,0 +1,31 @@
+### Create a new sub-project
+
+A new sub-project can be created as a repository and contributed to the Notary Project organization by opening a GitHub issue to request [Notary Project governance maintainers](MAINTAINERS) to vote in the
+[notaryproject/.github](https://github.com/notaryproject/.github) repository. A new sub-project can be created after having a supermajority of approval from Notary Project governance maintainers.
```
Is it a two-thirds supermajority or 60% supermajority? See https://en.wikipedia.org/wiki/Supermajority for common supermajorities. I would suggest two-thirds supermajority.

FeynmanZhou replied:
two-thirds is fair. I clarified it

yizha1 commented on REPOSITORIES.md:
```diff
@@ -0,0 +1,31 @@
+
+#### Unarchiviation
+
+Archived repositories can be proposed for unarchiviation by opening a GitHub issue in the [notaryproject/.github](https://github.com/notaryproject/.github) repository. Once the proposal issue has a supermajority of approval from Notary Project governance maintainers, the archived repository can be changed to active. In general, the same rules as for new repositories apply. 
```
Same comment on clarifying supermajority

yizha1 reviewed (commented):
Thanks @FeynmanZhou

shizhMSFT reviewed (commented):
What about deleting, renaming, and transferring out a repository?

FeynmanZhou commented:
> What about deleting, renaming, and transferring out a repository?

@shizhMSFT I added a section for Renaming. Deleting and transferring out a repository are not common cases so we could consider adding them in the future. Keep the content as minimum viable would be easier to review.

shizhMSFT commented on REPOSITORIES.md:
```diff
@@ -0,0 +1,36 @@
+
+#### Archiviation
+
+Repositories may become stable or stale, showing little to no activity during the time span of a year. They can be proposed for Archiviation by opening a GitHub [issue](https://github.com/notaryproject/.github/issues). The repository-level maintainers should be noticed about this archiviation issue. Once the proposal issue has a supermajority (two-thirds) of approval from Notary Project governanc ...
```
Why we need to archive a stable repo? You cannot even open issues in an archived repository. Suggesting removing the word `stable`.

FeynmanZhou replied:
Removed "stable" to avoid ambiguity.

shizhMSFT reviewed (approved):
LGTM

yizha1 reviewed (approved):
Thanks @FeynmanZhou LGTM

FeynmanZhou commented:
/cc @notaryproject/notaryproject-governance-maintainers to help review this PR.

priteshbandi reviewed (approved):
LGTM
